File: modules/wem-admin-ui/src/main/resources/web/admin/common/js/schema/metadata/MetadataSchemaName.ts
module api.schema.metadata {

    import ModuleKey = api.module.ModuleKey;

    export class MetadataSchemaName extends api.module.ModuleBasedName {

        constructor(name: string) {
            api.util.assertNotNull(name, "Metadata schema name can't be null");
            var parts = name.split(api.module.ModuleBasedName.SEPARATOR);
            super(ModuleKey.fromString(parts[0]), parts[1]);
        }
    }
}

File: modules/wem-api/src/main/java/com/enonic/wem/api/schema/metadata/MetadataSchemaName.java
package com.enonic.wem.api.schema.metadata;

import org.apache.commons.lang.StringUtils;

import com.enonic.wem.api.module.ModuleBasedName;
import com.enonic.wem.api.module.ModuleKey;

public final class MetadataSchemaName
    extends ModuleBasedName
{
    private MetadataSchemaName( final ModuleKey moduleKey, final String localName )
    {
        super( moduleKey, localName );
    }

    public static MetadataSchemaName from( final ModuleKey moduleKey, final String localName )
    {
        return new MetadataSchemaName( moduleKey, localName );
    }

    public static MetadataSchemaName from( final String metadataName )
    {
        final String moduleKey = StringUtils.substringBefore( metadataName, SEPARATOR );
        final String localName = StringUtils.substringAfter( metadataName, SEPARATOR );
        return new MetadataSchemaName( ModuleKey.from( moduleKey ), localName );
    }
}
